Core/include/Acts/Utilities/Intersection.hpp

```diff
@@ -94,6 +94,33 @@ struct Intersection {
     // The current one wins, no re-ordering
     return true;
   }
+
+  static bool navigationOrder(const Intersection<DIM>& aIntersection,
+                              const Intersection<DIM>& bIntersection) {
+    if ((aIntersection.status == Status::unreachable) &&
+        (bIntersection.status != Status::unreachable)) {
+      return false;
+    }
+    if ((aIntersection.status != Status::unreachable) &&
+        (bIntersection.status == Status::unreachable)) {
+      return true;
+    }
+    // both are reachable or onSurface now
+
+    auto a = aIntersection.pathLength;
+    auto b = bIntersection.pathLength;
+
+    if (a >= 0 && b >= 0) {
+      return a < b;
+    }
+    if (a <= 0 && b <= 0) {
+      return a > b;
+    }
+    if (a > 0) {  // b < 0
```

Comment:
Could you explain in a comment why we favor forward propagation even if the intersection path length `a` is very large?